dalli.gemspec

```diff
@@ -26,4 +26,5 @@ Gem::Specification.new do |s|
   }
 
   s.add_dependency 'logger'
+  s.add_dependency 'opentelemetry-api'
```

Comment:
I think fine for the shopify dalli, but for the open source dalli we might want to make it an optional dependency and it raises and error if you try to include or configure it and don't have this dependency.